Datahub.Core/Services/IDatahubAuditingService.cs

```diff
@@ -47,4 +48,29 @@ public interface IDatahubAuditingService
         /// <returns></returns>
         Task TrackException(Exception exception, params (string key, string value)[] details);
     }
+
+    public static class IDatahubAuditingServiceExtensions
+    {
+        /// <summary>
+        /// Saves the changes async and tracks the changes as data change events
+        /// </summary>
+        public static async Task TrackSaveChangesAsync(this DbContext dbContext, IDatahubAuditingService auditService)
+        {
+            dbContext.ChangeTracker.DetectChanges();
+
+            foreach (var entry in dbContext.ChangeTracker.Entries())
```

Comment:
Maybe we should put this section in a try/catch - we don't want the save to fail in case one of the display/formatting stuff fails?